src/main/java/org/jetbrains/bsp/bazel/server/BepServer.java

```diff
@@ -88,156 +87,122 @@ public void publishLifecycleEvent(
   @Override
   public StreamObserver<PublishBuildToolEventStreamRequest> publishBuildToolEventStream(
       StreamObserver<PublishBuildToolEventStreamResponse> responseObserver) {
-    return new StreamObserver<PublishBuildToolEventStreamRequest>() {
-
-      @Override
-      public void onNext(PublishBuildToolEventStreamRequest request) {
-        if (request
-            .getOrderedBuildEvent()
-            .getEvent()
-            .getBazelEvent()
-            .getTypeUrl()
-            .equals("type.googleapis.com/build_event_stream.BuildEvent")) {
-          handleEvent(request.getOrderedBuildEvent().getEvent());
-        } else {
-          //          System.out.println("Got this request " + request);
-        }
-        PublishBuildToolEventStreamResponse response =
-            PublishBuildToolEventStreamResponse.newBuilder()
-                .setStreamId(request.getOrderedBuildEvent().getStreamId())
-                .setSequenceNumber(request.getOrderedBuildEvent().getSequenceNumber())
-                .build();
-        responseObserver.onNext(response);
+    return new BepStreamObserver(this, responseObserver);
+  }
+
+  public TreeSet<Uri> getCompilerClasspath() {
+    return compilerClasspath;
+  }
+
+  public Map<String, String> getDiagnosticsProtosLocations() {
+    return diagnosticsProtosLocations;
+  }
+
+  public void collectDiagnostics(
```

Comment:
I believe that this function should return this collected Diagnostics, input argument (`files ToDiagnostics`) isn't the best idea.

Reply:
I wanted to change it like this at first too (after seeing the usage in BspServer) but I think it makes sense this way. In other places it's used to go over many sources and gather the diagnostics from each one (see its usage in a loop in processActionEvent).